allow for csr init

DieracDelta opened the pull request:
A small change to allow for csr initialization on mepc, mtval, and mtvec from the policy language. Note that I'm using the convention of CSR names are as they appear in the ISA in all caps. Note to get this to work, hope-src/policies/entities/ISA/riscv.yml also needs to be modified to contain these names. I can also start a PR on that repo if the naming convention seems reasonable.

ccasin commented:
Sorry that we haven't merged this yet - I'm looking through some old PRs now.  Is it still the case that this requires changes in hope-policies?  Are those change on a branch somewhere?

DieracDelta commented:
@ccasin I changed the naming convention for CSRs to use all caps and added in a couple extra CSRs. As a result, changes need to be made to the entities file and any policies that used mtvec. It makes more sense to me to have these names completely upper cased. I just pushed the necessary changes to the policies repo (see here: https://github.com/draperlaboratory/hope-policies/tree/csr-init-pr).

ccasin commented:
@DieracDelta thanks!  I think we're going to keep the old names (mostly because I'm afraid we won't catch all the places they appear).  @JohnPMerrill is going to take a look at this stuff and we should get it merged this week.

JohnPMerrill commented on validator/riscv/csr_list.h:
```diff
@@ -0,0 +1,257 @@
+#define CSR_MHPMCOUNTER29H 0xb9d
+#define CSR_MHPMCOUNTER30H 0xb9e
+#define CSR_MHPMCOUNTER31H 0xb9f
+#endif
```
Can you add `// RISCV_CSR_ENCODING_H` here to match the top of the header guard?

JohnPMerrill commented on validator/riscv/rv32_validator.cc:
```diff
@@ -123,6 +124,13 @@ rv32_validator_t::rv32_validator_t(meta_set_cache_t *ms_cache,
+  // set initial tags for specific CSRs
+  ms = ms_factory->get_meta_set("ISA.RISCV.CSR.MEPC");
+  csr_tags[CSR_MEPC] = m_to_t(ms);
+  ms = ms_factory->get_meta_set("ISA.RISCV.CSR.MTVAL");
```
As @ccasin mentioned, we want to keep the existing naming convention for CSRs. Can you change `MTVAL` and `MTVEC` to `MTVal` and `MTVec`?

DieracDelta replied:
I'm happy to change this back--what should I change mepc to?

DieracDelta commented:
Okay, I've made the aforementioned changes and pushed to pr-csr-init.
